### PR TITLE
Consolidation of notifications channels for neighbors team

### PR DESCRIPTION
**Pull request by akiioto**
**Description**

Changes proposed in this pull request:

- Migrate all notifications from kyma-prow-alerts to the new kyma-neighbors-alerts Slack channel.
- ...
- ...

**Related issue(s)**

### CODE REVIEW

File: prow/jobs/test-infra/prow-periodics.yaml [hunk 1]
```diff
@@ -283,6 +283,6 @@ periodics: # runs on schedule
             command:
               - "generic-autobumper"
             args:
-              - "--config=configs/autobump-config/test-infra-autobump-config.yaml"
+              - "--config=prow/autobump-config/test-infra-autobump-config.yaml"
```
Comment (Sawthis) [inline prow/jobs/test-infra/prow-periodics.yaml:286]: is this path correct after refactor of test infra?  I think it should be now here: https://github.com/kyma-project/test-infra/tree/main/configs/autobump-config

File: templates/data/prow-periodics-data.yaml [hunk 2]
```diff
@@ -206,11 +206,11 @@ templates:
                     owner: neighbors
                     description: "Autobump image versions in prow jobs"
                   cron: "45 * * * 1-5"
-                  slack_channel: kyma-prow-alerts
+                  slack_channel: kyma-neighbors-alerts
                   image: europe-docker.pkg.dev/kyma-project/prod/k8s-prow/generic-autobumper:v20231030-70c8254b
                   command: generic-autobumper
                   args:
-                    - --config=configs/autobump-config/test-infra-autobump-config.yaml
+                    - --config=prow/autobump-config/test-infra-autobump-config.yaml
```
Comment (Sawthis) [inline templates/data/prow-periodics-data.yaml:213]: is this path correct after refactor of test infra?  I think it should be now here: https://github.com/kyma-project/test-infra/tree/main/configs/autobump-config

**Review by Sawthis — CHANGES_REQUESTED** [review]

File: prow/jobs/test-infra/prow-periodics.yaml (hunk 1 above)
Comment (akiioto) [reply prow/jobs/test-infra/prow-periodics.yaml]: corrected

File: templates/data/prow-periodics-data.yaml (hunk 2 above)
Comment (akiioto) [reply templates/data/prow-periodics-data.yaml]: corrected

**Review by Sawthis — APPROVED** [review]

**Pull request merged**